File: demos/retrofit-demo-java/app/src/main/java/com/smd/retrofitdemo/PeopleInSpaceAdapter.java
package com.smd.retrofitdemo;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import com.smd.retrofitdemo.network.Person;
import java.util.List;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import androidx.recyclerview.widget.RecyclerView.ViewHolder;

/**
 * Adapter for the list of astronauts
 */
class PeopleInSpaceAdapter extends RecyclerView.Adapter<ViewHolder> {

    List<Person> peopleInSpace;

    public PeopleInSpaceAdapter(List<Person> peopleInSpace) {
        this.peopleInSpace = peopleInSpace;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new PeopleInSpaceViewHolder(
                LayoutInflater.from(parent.getContext()).inflate(R.layout.list_item, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        ((PeopleInSpaceViewHolder) holder).bind(position);
    }

    @Override
    public int getItemCount() {
        return peopleInSpace.size();
    }

    /**
     * ViewHolder for showing the name of an astronaut and their craft
     */
    class PeopleInSpaceViewHolder extends ViewHolder {
        TextView name;
        TextView craft;

        public PeopleInSpaceViewHolder(View itemView) {
            super(itemView);
            name = itemView.findViewById(R.id.name);
            craft = itemView.findViewById(R.id.craft);
        }

        public void bind(int position) {
            Person astronaut = peopleInSpace.get(position);
            name.setText(astronaut.getName());
            craft.setText(astronaut.getCraft());
        }
    }

    public void replaceAll(List<Person> newData) {
        peopleInSpace.clear();
        peopleInSpace.addAll(newData);
        notifyDataSetChanged();
    }
}

File: lab5-demos/retrofit-demo-kotlin/app/src/main/java/com/smd/retrofitdemo/PeopleInSpaceAdapter.kt
package com.smd.retrofitdemo

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.smd.retrofitdemo.databinding.ListItemBinding
import com.smd.retrofitdemo.network.Person

/**
 * Adapter for the list of astronauts
 */
internal class PeopleInSpaceAdapter(var peopleInSpace: MutableList<Person>) :
    RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val binding = ListItemBinding
            .inflate(LayoutInflater.from(parent.context), parent, false)
        return PeopleInSpaceViewHolder(binding)
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        (holder as PeopleInSpaceViewHolder).bind(position)
    }

    override fun getItemCount(): Int {
        return peopleInSpace.size
    }

    /**
     * ViewHolder for showing the name of an astronaut and their craft
     */
    internal inner class PeopleInSpaceViewHolder(private val viewBinding: ListItemBinding) :
        RecyclerView.ViewHolder(viewBinding.root) {

        fun bind(position: Int) {
            val astronaut = peopleInSpace[position]
            viewBinding.name.text = astronaut.name
            viewBinding.craft.text = astronaut.craft
        }
    }

    fun replaceAll(newData: List<Person>?) {
        peopleInSpace.clear()
        peopleInSpace.addAll(newData!!)
        notifyDataSetChanged()
    }
}